packages/expect/src/__tests__/utils.test.js

```diff
@@ -202,6 +202,60 @@ describe('subsetEquality()', () => {
   test('undefined does not return errors', () => {
     expect(subsetEquality(undefined, {foo: 'bar'})).not.toBeTruthy();
   });
+
+  describe('matching subsets with circular references', () => {
+    test('simple circular references', () => {
+      const circularObj = {a: 'hello'};
+      circularObj.ref = circularObj;
+
+      const otherCircularObj = {a: 'hello'};
+      otherCircularObj.ref = otherCircularObj;
+
+      const differentCircularObj = {a: 'world'};
+      differentCircularObj.ref = differentCircularObj;
+
+      const primitiveInsteadOfRef = {};
+      primitiveInsteadOfRef.ref = 'not a ref';
+
+      expect(subsetEquality(circularObj, {})).toBe(true);
+      expect(subsetEquality({}, circularObj)).toBe(false);
+      expect(subsetEquality(otherCircularObj, circularObj)).toBe(true);
+      expect(subsetEquality(differentCircularObj, circularObj)).toBe(false);
+      expect(subsetEquality(primitiveInsteadOfRef, circularObj)).toBe(false);
+    });
+
+    test('transitive circular references', () => {
+      const transitiveCircularObj = {a: 'hello'};
+      transitiveCircularObj.nestedObj = {parentObj: transitiveCircularObj};
+
+      const otherTransitiveCircularObj = {a: 'hello'};
```

Comment:
To avoid prettier wrap and more easily see parallel assertions, what do you think about variable names like:

| from | to |
| :--- | :--- |
| `transitiveCircularObj` | `transitiveCircularObjA1` |
| `otherTransitiveCircularObj` | `transitiveCircularObjA2` |
| `differentTransitiveCircularObj` | `transitiveCircularObjB` |

And if it help readability even without wrap, then similar names for simple circular test above

Reply:
Hi @pedrottimark, thanks for the review 😊 

I think the names you have suggested make tests more comprehensible indeed. They don't help much with wrapping, though, but I think it's worth changing for the sake of clarity.